### DISCUSSION TITLE
What is the "pageProps.initialReduxState"?

### DISCUSSION THREAD

**Discussion by chkwak** [vercel/next.js#49574] [Help] [answered]
### Summary

In "with-react-thunk-app example", what is the "pageProps.initialReduxState"?
( const store = useStore(pageProps.initialReduxState) )
I can't find any pages declaring the intialReduxState.
And when I console log "pageProps.initialReduxState", it's undefined.
How can you define this variable?

### Additional information

```js
import { Provider } from 'react-redux'
import { useStore } from '../store'

export default function App({ Component, pageProps }) {
  
  const store = useStore(pageProps.initialReduxState)

  return (
    <Provider store={store}>
      <Component {...pageProps} />
    </Provider>
  )
}
```

### Example

https://github.com/vercel/next.js/tree/canary/examples/with-redux-thunk

**Comment by icyJoseph** [comment, marked as answer]
IIRC, it is something like, each page could inject their own redux store config. So in `getStaticProps`, or `getServerSideProps`, you could return a prop named `initialReduxState` - 

Otherwise, since it is undefined, the store is not re-initialised on page navigation.

```js
export function useStore(initialState) {
  const store = useMemo(() => initializeStore(initialState), [initialState]) // if initial state is constantly undefined, the store persist
  return store
}
```

**Reply by chkwak** [reply]
Oh. I restarted my project. It works again. Thank you.
